src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java

```diff
@@ -7,46 +7,116 @@
+  public void mercury_search_navigateToPageUsingSearchSuggestions() {
+    init();
+
+    Search search = this.topBar.openSearch().typeInSearch(SEARCH_PHRASE);
+    String oldUrl = driver.getCurrentUrl();
+    search.selectSearchSuggestion(0);
+
+    Assertion.assertFalse(oldUrl.equalsIgnoreCase(driver.getCurrentUrl()),
+                          "Navigation to selected search suggestion failed");
+    Assertion.assertTrue(driver.getCurrentUrl().contains(SEARCH_PHRASE));
+  }
+
+  @Test(groups = "mercury_search_navigateToPageUsingSearchResults")
+  public void mercury_search_navigateToPageUsingSearchResults() {
     init();
 
-    topBar.openSearch().typeInSearch(searchingPhrase).selectSearchSuggestion(0);
+    SearchResultsPage searchResultsPage = new SearchResultsPage().openForQuery(SEARCH_PHRASE);
+    String oldUrl = driver.getCurrentUrl();
+    searchResultsPage.selectSearchResult(0);
 
-    Assertion.assertTrue(driver.getCurrentUrl().contains(searchingPhrase));
+    Assertion.assertFalse(oldUrl.equalsIgnoreCase(driver.getCurrentUrl()),
+                          "Navigation to selected search result failed");
+    Assertion.assertTrue(driver.getCurrentUrl().contains(SEARCH_PHRASE));
   }
 
   @Test(groups = "mercury_search_cancelSearchPhrase")
   public void mercury_search_clearSearchPhrase() {
-    String searchingPhrase = "Infobox";
     init();
 
-    Search search = topBar.openSearch();
-    search.typeInSearch(searchingPhrase);
-    search.clickClearSearchButton();
+    this.topBar.openSearch().typeInSearch(SEARCH_PHRASE).clickClearSearchButton();
+
+    Assertion.assertEquals(this.getSearch().getSearchPhrase(), "");
+  }
+
+  @Test(groups = "mercury_search_verifySearchLayout")
+  public void mercury_search_verifySearchLayout() {
+    init();
+
+    Search searchInstance = this.topBar.openSearch();
+
+    Assertion.assertTrue(searchInstance.isSearchInputFieldVisible());
+    Assertion.assertTrue(searchInstance.isClearSearchButtonVisible());
+    Assertion.assertTrue(searchInstance.isInputFieldSearchIconVisible());
+  }
+
+  @Test(groups = "mercury_search_userIsRedirectedToSearchResultsPage")
+  public void mercury_search_userIsRedirectedToSearchResultsPage() {
+    init();
+
+    SearchResultsPage searchResults =
+        this.topBar.openSearch().typeInSearch(SEARCH_PHRASE)
+            .clickEnterAndNavigateToSearchResults();
+    Assertion.assertTrue(searchResults.isSearchResultsPageOpen());
+  }
+
+  @Test(groups = "mercury_search_searchResultsPageHasNoSearchIconInTopBar")
+  public void mercury_search_searchResultsPageHasNoSearchIconInTopBar() {
+    init();
+
+    new SearchResultsPage().openForQuery(SEARCH_PHRASE);
+    Assertion.assertFalse(this.topBar.isSearchIconClickable());
+  }
+
+  @Test(groups = "mercury_search_searchInputDoesNotCoverNavigation",
+      expectedExceptions = WebDriverException.class)
+  public void mercury_search_searchInputDoesNotCoverNavigation() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)